Modules/_io/textio.c

```diff
@@ -1696,7 +1710,7 @@ _io_TextIOWrapper_write_impl(textio *self, PyObject *text)
     }
 
     self->pending_bytes_count += bytes_len;
-    if (self->pending_bytes_count > self->chunk_size || needflush ||
+    if (self->pending_bytes_count >= self->chunk_size || needflush ||
```

Comment:
Good idea. I think it's sensible to flush `pending_bytes` as soon there's a complete chunk instead of waiting for the next write.